code/model/modelrender.cpp

```diff
@@ -3072,6 +3072,40 @@ void model_render_set_wireframe_color(const color* clr)
 	Wireframe_color = *clr;
 }
 
+void modelinstance_replace_active_texture(polymodel_instance* pmi, const char* old_name, const char* new_name)
```

Comment:
This should bail (and probably warn) if `pmi` is nullptr.  Ships always have a model instance, but other object types are not guaranteed to.

Reply:
I'd have imagined that that's the responsibility of the caller, as this function is never directly exposed. But an Assert here is probably correct then